crates/uv/src/cli.rs

```diff
@@ -99,6 +99,41 @@ pub(crate) struct GlobalArgs {
     /// parent directories.
     #[arg(global = true, long, hide = true)]
     pub(crate) isolated: bool,
+
+    /// The Python interpreter to use for the current command.
+    ///
+    /// By default, `uv` searches for an active virtual environment, followed by a virtual
+    /// environment in the current working directory or any parent directory. The `--python` option
+    /// allows you to specify a different interpreter, which is intended for use in continuous
+    /// integration (CI) environments or other automated workflows.
+    ///
+    /// Supported formats:
+    /// - `3.10` looks for an installed Python 3.10 using `py --list-paths` on Windows, or
+    ///   `python3.10` on Linux and macOS.
+    /// - `python3.10` or `python.exe` looks for a binary with the given name in `PATH`.
+    /// - `/home/ferris/.local/bin/python3.10` uses the exact Python at the given path.
+    #[arg(global = true, long, short, env = "UV_PYTHON", verbatim_doc_comment)]
+    pub(crate) python: Option<String>,
```

Comment:
I have similar opinion on https://github.com/astral-sh/uv/pull/3864:
> Add a `--package` option that allows switching the current project in the workspace. Wherever you are in a workspace, you should be able to run with any other project as root. This is the uv equivalent of `cargo run -p`.
> 
> I don't love the `--package` name, esp. since `-p` is already taken and in general to many things start with p already.


What I can suggest is to use `--py` alias instad of `-p` short at there.
for _short_, I think we could rename the args to `python_toolchain` and use `-t` for _short_ usage.

It's breaking but we can also continue without _short_ at this point.